api_app/db/repositories/workspaces.py

```diff
@@ -67,14 +67,18 @@ def create_workspace_item(self, workspace_input: WorkspaceInCreate, auth_info: d
 
         template = self.validate_input_against_template(workspace_input.templateName, workspace_input, ResourceType.Workspace, user_roles)
 
+        # allow for workspace template taking a single address_space or multiple address_spaces
         address_space_param = {"address_space": self.get_address_space_based_on_size(workspace_input.properties)}
+        address_spaces_param = {"address_spaces": [self.get_address_space_based_on_size(workspace_input.properties)]}
```

Comment:
Will this always validate that the value provided in the new field is free?
get_address_space_based_on_size is doing that internally, but only if address_space_size=custom. What will happen if the primary field will get assigned automatically (e.g. size=small)?

Reply:
This looks wrong, will take another look, it should be:

```
        address_spaces_param = {"address_spaces": [address_space_param]}
```